tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+            with self.assertRaises(MissingCoordinateVariable) as context:
+                lat_prefetch_arr, lon_prefetch_arr = get_lat_lon_arrays(
+                    prefetch_dataset,
+                    self.varinfo.get_variable(
+                        '/Soil_Moisture_Retrieval_Data_AM/latitude_centroid'
+                    ),
+                    self.varinfo.get_variable(
+                        '/Soil_Moisture_Retrieval_Data_AM/longitude_centroid'
+                    ),
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'Coordinate: "/Soil_Moisture_Retrieval_Data_AM/latitude_centroid" is '
+                    'not present in source granule file.',
+                )
+
+    def test_get_override_projected_dimension_names(self):
+        """Ensure that the expected projected dimension name
+        is returned for the coordinate variables
+        """
+
+        expected_projected_names = [
+            '/Soil_Moisture_Retrieval_Data_AM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_AM/projected_x',
+        ]
+
+        with self.subTest(
+            'Retrieves expected projected dimension names for a science variable'
+        ):
+            self.assertListEqual(
+                get_override_projected_dimension_names(self.varinfo, self.latitude),
+                expected_projected_names,
+            )
+
+        with self.subTest(
+            'Retrieves expected dimension names for the longitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimension_names(self.varinfo, self.longitude),
+                expected_projected_names,
+            )
+
+        with self.subTest('Raises exception for missing coordinate variable'):
+            with self.assertRaises(MissingCoordinateVariable) as context:
+                get_override_projected_dimension_names(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/longitude_centroid'
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'Coordinate: "/Soil_Moisture_Retrieval_Data_AM/latitude_centroid" is '
+                    'not present in source granule file.',
+                )
+
+    def test_get_override_projected_dimensions(self):
+        """Ensure that the expected projected dimension name
+        is returned for the coordinate variables
+        """
+
+        expected_override_dimensions_AM = [
+            '/Soil_Moisture_Retrieval_Data_AM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_AM/projected_x',
+        ]
+        expected_override_dimensions_PM = [
+            '/Soil_Moisture_Retrieval_Data_PM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_PM/projected_x',
+        ]
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the science variable'
+        ):
+            self.assertEqual(
```

Comment:
Nitpick: These subtests are all comparing `list` items, so you could/should use `self.assertListEqual` instead of just `self.assertEqual`.

Reply:
updated - 36a3c4034215d85f5fccf660dc3f213d28873d3d